test/lit/passes/type-generalizing.wast

```diff
@@ -0,0 +1,219 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py --all-items and should not be edited.
+;; RUN: foreach %s %t wasm-opt --dce --experimental-type-generalizing -all -S -o - | filecheck %s
+
+(module
+
+ ;; CHECK:      (type $0 (func))
+
+ ;; CHECK:      (type $1 (func (result eqref)))
+
+ ;; CHECK:      (type $2 (func (param anyref)))
+
+ ;; CHECK:      (type $3 (func (param i31ref)))
+
+ ;; CHECK:      (type $4 (func (param anyref eqref)))
+
+ ;; CHECK:      (type $5 (func (param eqref)))
+
+ ;; CHECK:      (func $unconstrained (type $0)
+ ;; CHECK-NEXT:  (local $x i32)
+ ;; CHECK-NEXT:  (local $y anyref)
+ ;; CHECK-NEXT:  (local $z (anyref i32))
+ ;; CHECK-NEXT:  (nop)
+ ;; CHECK-NEXT: )
+ (func $unconstrained
+  ;; This non-ref local should be unmodified
+  (local $x i32)
+  ;; There is no constraint on the type of this local, so make it top.
```

Comment:
```suggestion
  ;; There is no constraint on the type of this local, so leave it as top.
```

Reply:
Oh, I think the `anyref` below is supposed to be something else.